### ISSUE TITLE
Gradle 4.10.1+ is unable to build a working war project within a ear project if scala plugin is used

### ISSUE THREAD

**Issue by KrauseStefan** [gradle/gradle#7014]
Gradle 4.10.1 is unable to build a sound war project within a ear project if Scala plugin is used.

Projects structure:
```
build.gradle (parrent)
|- build.gralde (war)
\- build.gralde (ear)
```
We have 200+ Scala sub projects in the overall project but only 2 sub projects is needed to reproduce.

All projects are configured to apply the Scala plugin from the parrent project

### Expected Behavior
I would expect the behavior to be simuler to the one in previous Gradle versions

### Current Behavior
The ear file contains a unexpected `compileScala.mapping` file, and the file has an entry in `applications.xml`

``` xml
  <module>
    <ejb>compileScala.mapping</ejb>
  </module>
```
This causes in our case Wildfly to try to deploy the `compileScala.mapping` and fails at the extraction step.

### Steps to Reproduce (for bugs)
* Clone this repository: https://github.com/KrauseStefan/gradle_4.10.2_scala_issue
* Build
* Deploy to Wildfly (probably also other applications servers)

### Your Environment
Windows 10 1709

I've tested with a few Gradle version but the latest fails also fails
```
$  .\gradlew.bat -version
------------------------------------------------------------
Gradle 4.10.2
------------------------------------------------------------

Build time:   2018-09-19 18:10:15 UTC
Revision:     b4d8d5d170bb4ba516e88d7fe5647e2323d791dd

Kotlin DSL:   1.0-rc-6
Kotlin:       1.2.61
Groovy:       2.4.15
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          1.8.0_171 (Oracle Corporation 25.171-b11)
OS:           Windows 10 10.0 amd64
```

We are under strict rules not to leak source code.
If I cannot inspect what a build scan publishes I cannot share it, sorry.

**Comment by KrauseStefan** [comment]
Link to build output: https://github.com/KrauseStefan/gradle_4.10.2_scala_issue/raw/master/ear/build/libs/ear.ear

**Comment by oehme** [comment]
@big-guy Can you look into this one please?

**Comment by KrauseStefan** [comment]
I see that the issue was added to the 5.0 RC1, does this mean that it will not be fixed for 4.10?
We are currently using 4.7 and I wanted to stop at 4.10 before fixing the deprecation warnings that would prevent us moving to 5.0

**Comment by big-guy** [comment]
When I have a better idea of the fix for the underlying problem, we'll see about backporting.

As a workaround, can you try adding this to your `war` projects?
```
configurations.archives.artifacts.clear()
artifacts {
   archives war
}
```

**Comment by KrauseStefan** [comment]
Thank you I will try that tomorrow and get back to you

**Comment by KrauseStefan** [comment]
The workaround works like a charm.
Thank you, I guess we should wait with closing this until the fixes are released.

**Comment by big-guy** [comment]
This will be released in 4.10.3 (hopefully today)
